### ISSUE TITLE
Fails on Node.js v12

### ISSUE THREAD

**Issue by BridgeAR** [kelektiv/node.bcrypt.js#715]
A couple fixes landed to make `bcrypt` compatible with Node.js v12 but it seems like it still fails to build (https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/1801/nodes=debian9-64/testReport/junit/(root)/citgm/bcrypt_v3_0_5/).

Refs: https://github.com/nodejs/citgm/pull/701

**Comment by richardlau** [comment]
The recent breakage was (intentionally) caused by https://github.com/nodejs/node/pull/27013.

`nan` needs to be 2.13.2 to pick up https://github.com/nodejs/nan/pull/842:
https://github.com/nodejs/nan/blob/master/CHANGELOG.md

**Comment by agathver** [comment]
@richardlau v3.0.6 is out with updated nan

**Comment by YasharF** [comment]
Is this still a problem?

**Comment by BridgeAR** [comment]
@YasharF this should be fixed with 3.0.6.

**Comment by NicTanghe** [comment]
I get the following error when trying to install

```
node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp WARN Tried to download(404): https://github.com/kelektiv/node.bcrypt.js/releases/download/v3.0.6/bcrypt_lib-v3.0.6-node-v72-win32-x64-unknown.tar.gz
node-pre-gyp WARN Pre-built binaries not found for bcrypt@3.0.6 and node@12.9.1 (node-v72 ABI, unknown) (falling back to source compile with node-gyp)
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  blowfish.cc
  bcrypt.cc
  bcrypt_node.cc
  win_delay_load_hook.cc
C:\Users\nicta\AppData\Local\node-gyp\Cache\12.9.1\x64\node.lib : fatal error LNK1127: library is corrupt [C:\Users\nic ta\Documents\dev\node\vidly-api-node\node_modules\bcrypt\build\bcrypt_lib.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:209:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "build" "--fallback-to-build" "--module=C:\\Users\\nicta\\Documents\\dev\\node\\vidly-api-node\\node_modules\\bcrypt\\lib\\binding\\bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=C:\\Users\\nicta\\Documents\\dev\\node\\vidly-api-node\\node_modules\\bcrypt\\lib\\binding" "--napi_version=4" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
gyp ERR! cwd C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt
gyp ERR! node -v v12.9.1
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt\lib\binding --napi_version=4 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:209:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
node-pre-gyp ERR! System Windows_NT 10.0.18362
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\nicta\\Documents\\dev\\node\\vidly-api-node\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt
node-pre-gyp ERR! node -v v12.9.1
node-pre-gyp ERR! node-pre-gyp -v v0.12.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\nicta\Documents\dev\node\vidly-api-node\node_modules\bcrypt\lib\binding --napi_version=4 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
npm WARN vidly@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bcrypt@3.0.6 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bcrypt@3.0.6 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\nicta\AppData\Roaming\npm-cache\_logs\2019-08-28T13_34_23_836Z-debug.log
C:\Users\nicta\Documents\dev\node\vidly-api-node [master ≡ +0 ~3 -0 !]
```


[2019-08-28T13_34_23_836Z-debug.log](https://github.com/kelektiv/node.bcrypt.js/files/3551148/2019-08-28T13_34_23_836Z-debug.log)
